pytorch_lightning/trainer/training_loop.py

```diff
@@ -461,6 +461,11 @@ def run_training_epoch(self):
 
         # modify dataloader if needed (ddp, etc...)
         train_dataloader = self.prepare_train_loop_dataloader(self.train_dataloader)
+        try:
+            self.train_dataloader_len = len(train_dataloader)
+        except (NotImplementedError, TypeError):
+            # for infinite train_dataloader
+            self.train_dataloader_len = 0
```

Comment:
you can use `self.num_training_batches` so that it will work with `limit_train_batches` too.
```suggestion
```
you can use `self.num_training_batches` so that it will work with `limit_train_batches` too.

Reply:
and can we use the `has_len` helper function which is already defined somewhere in the trainer, because there is an important edge case implemented, where in iterable dataset actually can have a length.

Reply:
Got it! I didn't notice that it was already defined.